packages/dashboards/addon/services/dashboard-data.js

```diff
@@ -1,89 +1,155 @@
    */
-  fetchDataForDashboard(dashboard) {
-    return dashboard
-      .get('widgets')
-      .then(widgets => this.fetchDataForWidgets(dashboard.id, widgets, [], get(this, 'widgetOptions')));
-  },
+  async fetchDataForDashboard(dashboard) {
+    const widgets = await dashboard.widgets;
+    const layout = get(dashboard, 'presentation.layout');
+
+    return this.fetchDataForWidgets(dashboard.id, widgets, layout, [], this.widgetOptions);
+  }
 
   /**
    * @method fetchDataForWidgets
+   * @param {Number} dashboardId
    * @param {Array} widgets - list of widget models with requests to fetch
+   * @param {Array} layout - dashboard layout
    * @param {Array} decorators - array of functions to modify each request
    * @param {Object} options - options for web service fetch
-   * @returns {Object} hash of widget id to data promise array
+   * @returns {Object} hash of the widget id to its TaskInstance
    */
-  fetchDataForWidgets(dashboardId, widgets = [], decorators = [], options = {}) {
-    const result = {};
-    const uuid = v1();
-
-    // Construct hash of widget id to data
-    widgets.forEach(widget => {
-      const widgetId = get(widget, 'id');
-      const dashboard = get(widget, 'dashboard');
-      const widgetDataPromises = get(widget, 'requests').map(request => {
-        //construct custom header for each widget with uuid
-        options.customHeaders = {
-          uiView: `dashboard.${dashboardId}.${uuid}.${widgetId}`
-        };
-
-        const requestWithFilters = this._applyFilters(dashboard, request);
-        const requestDecorated = this._decorate(decorators, requestWithFilters.serialize());
-
-        const filterErrors = this._getFilterErrors(dashboard, request);
-
-        return this._fetch(requestDecorated, options).then(result => {
-          const serverErrors = getWithDefault(result, 'response.meta.errors', []);
-
-          return merge({}, result, { response: { meta: { errors: [...serverErrors, ...filterErrors] } } });
-        });
+  fetchDataForWidgets(dashboardId, widgets = [], layout = [], decorators = [], options = {}) {
+    const uuid = v1(),
+      taskByWidget = {};
+
+    // sort widgets by order in layout
+    const sortedWidgets = arr(layout)
+      .sortBy('row', 'column')
+      .map(layoutItem => widgets.find(widget => widget.id == layoutItem.widgetId))
+      .filter(widget => widget);
+
+    sortedWidgets.forEach(widget => {
+      const taskInstance = this._fetchRequestsForWidget.perform(dashboardId, widget, decorators, options, uuid);
+
+      taskByWidget[widget.id] = taskInstance;
+
+      /**
+       * bubbling 403 errors causes acceptance test to fail https://github.com/emberjs/ember-qunit/issues/592
+       * task still expectedly fails.
+       * TODO: no need for catch block when ^ resolves (needlessly catches TaskCancelation errors as well)
+       */
+      taskInstance.catch(error => {
+        if (isForbidden(error) || (error && error.name === 'TaskCancelation')) {
```

Comment:
Do we still get forbidden messages in the UI?

Reply:
Yes we still check `{{#if (is-forbidden @taskInstance.error)}}`